draft-js-plugins-editor/src/Editor/index.js

```diff
@@ -139,165 +183,197 @@ class PluginEditor extends Component {
+    let styleMaps = [];
+    if (plugins != null) {
+      styleMaps = plugins
+        .filter((plug) => plug.customStyleMap != null)
+        .map((plug) => plug.customStyleMap);
+    }
 
-    eventHookKeys.forEach((attrName) => {
-      pluginHooks[attrName] = this.createEventHooks(attrName, plugins);
-    });
+    if (customStyleMap != null) {
+      styleMaps = styleMaps.concat([customStyleMap]);
+    }
 
-    handleHookKeys.forEach((attrName) => {
-      pluginHooks[attrName] = this.createHandleHooks(attrName, plugins);
-    });
+    return styleMaps.reduce((styles, style) => ({ ...styles, ...style, }), {});
+  }
 
-    fnHookKeys.forEach((attrName) => {
-      pluginHooks[attrName] = this.createFnHooks(attrName, plugins);
-    });
+  resolveblockRenderMap = (): DraftBlockRenderMap => {
+    const { defaultBlockRenderMap, blockRenderMap, plugins } = this.props;
 
-    return pluginHooks;
-  };
+    let map = Map({});
 
-  resolvePlugins = () => {
-    const plugins = this.props.plugins.slice(0);
-    if (this.props.defaultKeyBindings) {
-      plugins.push(defaultKeyBindingPlugin);
+    /*
+     * TODO: Add this to docs probably doctypes
+     * props.blockRenderMap > plugin.blockRenderMap  > DefaultDraftBlockRenderMap
+     */
+    if (defaultBlockRenderMap === true) {
+      map = map.merge(DefaultDraftBlockRenderMap);
     }
 
-    return plugins;
-  };
-
-  resolveCustomStyleMap = () => (
-    this.props.plugins
-      .filter((plug) => plug.customStyleMap !== undefined)
-      .map((plug) => plug.customStyleMap)
-      .concat([this.props.customStyleMap])
-      .reduce((styles, style) => (
-        {
-          ...styles,
-          ...style,
-        }
-      ), {})
-  );
-
-  resolveblockRenderMap = () => {
-    let blockRenderMap = this.props.plugins
-      .filter((plug) => plug.blockRenderMap !== undefined)
-      .reduce((maps, plug) => maps.merge(plug.blockRenderMap), Map({}));
-    if (this.props.defaultBlockRenderMap) {
-      blockRenderMap = DefaultDraftBlockRenderMap.merge(blockRenderMap);
+    if (plugins != null) {
+      map = plugins
+        .reduce((styles, plug) => (
+          plug.blockRenderMap != null
+            ? styles.merge(plug.blockRenderMap)
+            : styles), map);
     }
-    if (this.props.blockRenderMap) {
-      blockRenderMap = blockRenderMap.merge(this.props.blockRenderMap);
+
+    if (blockRenderMap != null) {
+      map = map.merge(blockRenderMap);
     }
-    return blockRenderMap;
+
+    return map;
   }
 
   resolveAccessibilityProps = () => {
     let accessibilityProps = {};
-    const plugins = [this.props, ...this.resolvePlugins()];
+    const plugins = this.resolvePlugins();
+
     plugins.forEach((plugin) => {
       if (typeof plugin.getAccessibilityProps !== 'function') return;
       const props = plugin.getAccessibilityProps();
       const popupProps = {};
 
-      if (accessibilityProps.ariaHasPopup === undefined) {
+      if (accessibilityProps.ariaHasPopup == null) {
```

Comment:
Is this every true? I guess most of the times it's undefined and that's why we had the check for it?!

Reply:
`value == null` is a shorthand for `value === null || value === undefined` :+1: #javascript

Reply:
ah missed the `==` … can you add that as a comment on top of it to make it beginner friendly and people like me constantly forgetting this :)